### PR TITLE
V2.0 add stat.tbl continuous tests

### PR DESCRIPTION
**Pull request by zdz2101**
**What changes are proposed in this pull request?**

**If there is an GitHub issue associated with this pull request, please provide link.**

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [x] Ensure all package dependencies are installed by running `renv::install()`
- [x] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [x] If an update was made to `tbl_summary()`, was the same change implemented for `tbl_svysummary()`?
- [x] If a new function was added, function included in `_pkgdown.yml`
- [x] If a bug was fixed, a unit test was added for the bug check
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Code coverage is suitable for any new functions/features. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Begin in a fresh R session without any packages loaded. 
- [x] R CMD Check runs without errors, warnings, and notes
- [x] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [x] Update `NEWS.md` with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [x] Increment the version number using `usethis::use_version(which = "dev")` 
- [x] Run `codemetar::write_codemeta()`
- [x] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: R/add_stat.R [hunk 1]
```diff
@@ -148,6 +148,8 @@ add_stat <- function(x, fns, location = everything() ~ "label") {
     dplyr::tibble(variable = names(fns)) |>
     dplyr::mutate(
       summary_type = map_chr(.data$variable, ~ x$inputs$type[[.x]]),
```
Comment (zdz2101) [inline R/add_stat.R:150]: @ddsjoberg this was causing a break in even the original unit tests for `add_stat()` for `tbl_continuous()` objects, so I made a modification to `tbl_continuous.R` to at least get it "back to working" as it was previously

File: R/add_stat.R [hunk 2]
```diff
@@ -148,6 +148,8 @@ add_stat <- function(x, fns, location = everything() ~ "label") {
...
+      # tbl_continuous objects generally didn't have a "$inputs$type" object nested inside
```
Comment (zdz2101) [inline R/add_stat.R:151]: explanation is here, `tbl_continuous` objects didn't have `x$inputs$type` inside them

File: R/tbl_continuous.R [hunk 3]
```diff
@@ -94,6 +95,7 @@ tbl_continuous <- function(data,
     digits = digits
   )
 
+  type <- assign_summary_type(data, variables = include, value = NULL)
```
Comment (zdz2101) [inline R/tbl_continuous.R:98]: by restoring it

File: tests/testthat/test-add_stat.tbl_continuous.R [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  add_stat_test1 <- function(data, variable, by, ...) {
+    tibble::tibble(test_col = "Ugh")
+  }
+
+  add_stat_test2 <- function(data, variable, by, ...) {
+    tibble::tibble(test_col = rep_len("Ugh", 3))
+  }
+
+  expect_error(
```
Comment (zdz2101) [inline tests/testthat/test-add_stat.tbl_continuous.R:18]: these work again

File: R/add_stat.R [hunk 5]
```diff
@@ -231,6 +233,7 @@ eval_fn_safe <- function(variable, tbl, fn) {
         data = tbl$inputs$data,
         variable = variable,
         by = tbl$inputs$by,
+        include = tbl$inputs$include,
```
Comment (zdz2101) [inline R/add_stat.R:236]: but I'm wondering if we should include tests similar to your `my_ttest()` examples for add_stat for tbl_summary objects, but I believe for the most part we would want access to the `include()` for the function formation, currently stuck/may need some assistance in this department

File: R/tbl_continuous.R (hunk 3 above)
Comment (zdz2101) [reply R/tbl_continuous.R:98]: type<- rep_named(x$inputs$include, list("categorical")) use something along these lines

File: tests/testthat/test-add_stat.tbl_continuous.R [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    round(3) %>%
+    as.character
+
+  add_stat_test2 <- function(data, variable, by, tbl, ...) {
+    col_name <- tbl$inputs$include
+    col_sym <- sym(col_name)
+
+    strat <- dplyr::group_split(data, !!col_sym)
+
+    lapply(strat, function(x) t.test(x$age ~ x$trt)$p.value) %>% unlist
```
Comment (zdz2101) [inline tests/testthat/test-add_stat.tbl_continuous.R:47]: @ddsjoberg do you know how I'd properly use variable and by in this scenario to "properly" complete this example?

File: R/tbl_continuous.R [hunk 7]
```diff
@@ -94,6 +94,7 @@ tbl_continuous <- function(data,
     digits = digits
   )
 
+  type <- rlang::rep_named(include, list("categorical"))
```
Comment (zdz2101) [inline R/tbl_continuous.R:97]: @ddsjoberg apparently this also breaks `add_p()` ?!? Can't say I can find the relationship why there

Comment (zdz2101) [reply R/tbl_continuous.R]: nvm it just needed to be modffied and slotted elsewhere

File: tests/testthat/test-add_stat.tbl_continuous.R [hunk 8]
```diff
@@ -0,0 +1,57 @@
+test_that("add_stat() for 'tbl_continuous'", {
```
Comment (ddsjoberg) [inline tests/testthat/test-add_stat.tbl_continuous.R:1]: Can we add these tests to `test-add_stat()` this name makes it look like an S3 method

File: tests/testthat/test-add_stat.tbl_continuous.R [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  tt <-
```
Comment (ddsjoberg) [inline tests/testthat/test-add_stat.tbl_continuous.R:2]: Where we can (which i think is all places), can we swap `%>%` for the base R pipe?

File: tests/testthat/test-add_stat.tbl_continuous.R (hunk 6 above)
Comment (ddsjoberg) [reply tests/testthat/test-add_stat.tbl_continuous.R:47]: i made the update in the file FYI

**Review by ddsjoberg — CHANGES_REQUESTED** [review]
Thank you @zdz2101 ! I think a couple of cosmetic changes are we're good!

FYI, I removed the type from `x$inputs` in `tbl_continuous()` since that is later called in `add_overall()` and we use those inputs as the arguments.

**Review by ddsjoberg — APPROVED** [review]
looks great, thank you! 🌟🌟🌟

**Pull request merged**